File: ApiProject/API-TFG/src/main/java/com/aitor/api_tfg/auth/ValidResponse.java
package com.aitor.api_tfg.auth;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ValidResponse {
    private Boolean isValid;
    private String message;
}

File: AndroidProject/app/src/main/java/com/aitor/trackactividades/authentication/data/response/ValidResponse.kt
package com.aitor.trackactividades.authentication.data.response

import com.aitor.trackactividades.authentication.presentation.model.ValidateTokenModel
import com.google.gson.annotations.SerializedName

data class ValidResponse(
    @SerializedName("isValid") val isValid: Boolean,
    @SerializedName("message") val message: String
) {
    fun toPresentation(): ValidateTokenModel {
        return ValidateTokenModel(
            isValidate = isValid,
            message = message
        )
    }
}
